### ISSUE TITLE
running forward loop using get_peft_model disables requires_grad on output

### ISSUE THREAD

**Issue by Hamidreza3252** [huggingface/peft#2410]
Hi, 
I would like to report a recent issue I have been facing, but I am not sure if it is a bug or I am doing something wrong in the process. The steps to re-create the steps are easy. The issue happens when I try to convert **Qwen2-VL-2B-Instruct** model into a PEFT model using `get_peft_model` method. Simply load the model using the sample code in https://huggingface.co/Qwen/Qwen2-VL-2B-Instruct and try to convert it to a PEFT model using a typical **8bit** LoraConfig with just sample `target_modules=["q_proj", "v_proj"]`. Then simply run a forward call to the model using a dummy input, such as `input_ids = torch.zeros((4, 1247)).to(device)`. When I inspect the `requires_grad` of `logits` attribute of the output, it is False. Meaning that I cannot run backward based on that output. This issue has been puzzling me for a while. I would appreciate if you can help me with a solution or advice how to address it properly.

**Comment by BenjaminBossan** [comment]
Hi, I can't reproduce the issue based on your description, could you please provide a reproducer?

```python
import torch
from transformers import Qwen2VLForConditionalGeneration, BitsAndBytesConfig
from peft import LoraConfig, get_peft_model

quant_config = BitsAndBytesConfig(load_in_8bit=True)
model = Qwen2VLForConditionalGeneration.from_pretrained(
    "Qwen/Qwen2-VL-2B-Instruct", device_map=0, quantization_config=quant_config,
)
model = get_peft_model(model, LoraConfig(target_modules=["q_proj", "v_proj"]))
input_ids = torch.zeros((4, 1247)).to(model.device).long()
output = model(input_ids)
assert output.logits.requires_grad is True
```

**Comment by Hamidreza3252** [comment]
I could find the source of discrepancies. Here are more details on my steps (sorry that I missed to mention that in my post above) 
I am trying to run https://github.com/EvolvingLMMs-Lab/open-r1-multimodal/tree/main using its bash, located in `local_scripts` folder. When I use `zero3.json` everything is fine, and my results are aligned with your. However, when I use `zero3_offload.json`, I get `requires_grad` as False. `zero3_offload.json` is supposed to offload some optimizers' parameters onto CPU so that GPU VRAM is offloaded. It seems that PEFT package detaches the tensor from the computational graph using this config. In order to reproduce it, you can comply run your conde above using torch-run in that repo.
